[libsystemd] Build error on x64-linux

Package: libsystemd:x64-linux@255#2

**Host Environment**

- Host: x64-linux
- Compiler: GNU 14.1.1
-    vcpkg-tool version: 2024-06-10-02590c430e4ed9215d27870138c2e579cc338772
    vcpkg-scripts version: 67a300519 2024-06-17 (19 hours ago)

**To Reproduce**

`vcpkg install `

**Failure logs**

```
-- Found Python version '3.12.3 at /usr/bin/python3'
-- Using meson: /home/suraj/vcpkg/downloads/tools/meson-1.3.2-6f83ac/meson.py
-- Using cached systemd-systemd-v255.tar.gz.
-- Cleaning sources at /home/suraj/vcpkg/buildtrees/libsystemd/src/v255-948a7f2e81.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/suraj/vcpkg/downloads/systemd-systemd-v255.tar.gz
-- Applying patch disable-warning-nonnull.patch
-- Applying patch only-libsystemd.patch
-- Applying patch pkgconfig.patch
-- Using source at /home/suraj/vcpkg/buildtrees/libsystemd/src/v255-948a7f2e81.clean
-- Found external ninja('1.12.1').
-- Configuring x64-linux-dbg
-- Getting CMake variables for x64-linux-dbg
-- Getting CMake variables for x64-linux-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: /usr/bin/python3 -I /home/suraj/vcpkg/downloads/tools/meson-1.3.2-6f83ac/meson.py setup -Dmode=release -Dstatic-libsystemd=pic -Dtests=false -Ddns-over-tls=false -Dtranslations=false -Dacl=disabled -Dapparmor=disabled -Daudit=disabled -Dblkid=disabled -Dbpf-framework=disabled -Dbzip2=disabled -Ddbus=disabled -Delfutils=disabled -Dfdisk=disabled -Dgcrypt=disabled -Dglib=disabled -Dgnutls=disabled -Dkmod=disabled -Dlibcurl=disabled -Dlibcryptsetup=disabled -Dlibfido2=disabled -Dlibidn=disabled -Dlibidn2=disabled -Dlibiptc=disabled -Dmicrohttpd=disabled -Dopenssl=disabled -Dp11kit=disabled -Dpam=disabled -Dpcre2=disabled -Dpolkit=disabled -Dpwquality=disabled -Dpasswdqc=disabled -Dseccomp=disabled -Dselinux=disabled -Dtpm2=disabled -Dxenctrl=disabled -Dxkbcommon=disabled -Dzlib=disabled -Dlz4=enabled -Dxz=enabled -Dzstd=enabled --backend ninja --wrap-mode nodownload -Doptimization=plain --native /home/suraj/vcpkg/buildtrees/libsystemd/meson-x64-linux-dbg.log --libdir lib --pkgconfig.relocatable -Ddebug=true --prefix /home/suraj/vcpkg/packages/libsystemd_x64-linux/debug --includedir ../include --pkg-config-path ['/home/suraj/work/test/build/vcpkg_installed/x64-linux/debug/lib/pkgconfig','/home/suraj/work/test/build/vcpkg_installed/x64-linux/share/pkgconfig'] -Dcmake_prefix_path=['/home/suraj/work/test/build/vcpkg_installed/x64-linux/debug','/home/suraj/work/test/build/vcpkg_installed/x64-linux'] /home/suraj/vcpkg/buildtrees/libsystemd/src/v255-948a7f2e81.clean
    Working Directory: /home/suraj/vcpkg/buildtrees/libsystemd/x64-linux-dbg
    Error code: 1
    See logs for more information:
      /home/suraj/vcpkg/buildtrees/libsystemd/config-x64-linux-dbg-meson-log.txt.log
      /home/suraj/vcpkg/buildtrees/libsystemd/config-x64-linux-dbg-out.log

Call Stack (most recent call first):
  /home/suraj/work/test/build/vcpkg_installed/x64-linux/share/vcpkg-tool-meson/vcpkg_configure_meson.cmake:463 (vcpkg_execute_required_process)
  ports/libsystemd/portfile.cmake:17 (vcpkg_configure_meson)
  scripts/ports.cmake:191 (include)



```

Please find the logs in attached files.
[config-x64-linux-dbg-meson-log.txt.log](https://github.com/user-attachments/files/15891001/config-x64-linux-dbg-meson-log.txt.log)
[config-x64-linux-dbg-out.log](https://github.com/user-attachments/files/15891003/config-x64-linux-dbg-out.log)

**Additional context**

<details><summary>vcpkg.json</summary>

```
{
  "name": "test",
  "dependencies": [
    "wxwidgets"
  ]
}

```
</details>

--- a/ports/libsystemd/only-libsystemd.patch
+++ b/ports/libsystemd/only-libsystemd.patch
@@ -1,8 +1,8 @@
 diff --git a/meson.build b/meson.build
-index 7419e2b..687450e 100644
+index b1a110c..6a7d930 100644
 --- a/meson.build
 +++ b/meson.build
-@@ -1909,12 +1909,11 @@ libsystemd_includes = [basic_includes, include_directories(
+@@ -2074,12 +2074,11 @@ libsystemd_includes = [basic_includes, include_directories(
  includes = [libsystemd_includes, include_directories('src/shared')]
  
  subdir('po')
@@ -17,7 +17,7 @@ index 7419e2b..687450e 100644
  
  libsystemd = shared_library(
          'systemd',
-@@ -1930,7 +1929,8 @@ libsystemd = shared_library(
+@@ -2095,7 +2094,8 @@ libsystemd = shared_library(
                          threads,
                          userspace],
          link_depends : libsystemd_sym,
@@ -27,16 +27,16 @@ index 7419e2b..687450e 100644
          install_tag: 'libsystemd',
          install_dir : libdir)
  
-@@ -1963,6 +1963,8 @@ install_libsystemd_static = static_library(
-                         userspace],
-         c_args : libsystemd_c_args + (static_libsystemd_pic ? [] : ['-fno-PIC']))
+@@ -2130,6 +2130,8 @@ else
+         alias_target('libsystemd', libsystemd)
+ endif
  
 +if false
 +
  libudev = shared_library(
          'udev',
          version : libudev_version,
-@@ -1974,7 +1976,8 @@ libudev = shared_library(
+@@ -2141,7 +2143,8 @@ libudev = shared_library(
          dependencies : [threads,
                          userspace],
          link_depends : libudev_sym,
@@ -46,9 +46,9 @@ index 7419e2b..687450e 100644
          install_tag: 'libudev',
          install_dir : libdir)
  
-@@ -2661,6 +2664,13 @@ if not meson.is_cross_build()
-                 command : [export_dbus_interfaces_py, '@OUTPUT@', dbus_programs])
- endif
+@@ -2905,6 +2908,13 @@ custom_target('installed-unit-files.txt',
+               install_dir : testdata_dir,
+               command : [meson_extract_unit_files, project_build_root])
  
 +endif # false
 +static_libudev='unused'
@@ -57,6 +57,6 @@ index 7419e2b..687450e 100644
 +want_html=false
 +want_man=false
 +
- ############################################################
+ #####################################################################
  
  alt_time_epoch = run_command('date', '-Is', '-u', '-d', '@@0@'.format(time_epoch),

--- a/ports/libsystemd/portfile.cmake
+++ b/ports/libsystemd/portfile.cmake
@@ -2,7 +2,7 @@ vcpkg_from_github(
   OUT_SOURCE_PATH SOURCE_PATH
   REPO systemd/systemd
   REF "v${VERSION}"
-  SHA512 51728de604c2169d8643718ac72acb8f70f613cfcca9e9abb7dac519f291fa26a16d48f24cae6897356319096cfe8f4d9377743e7870127374f98d432e0c557c
+  SHA512 10da82ee58d3608c41cb0204fdf0227af965b13b8f3716e4f5dea994c236c08a5e31f09ba0d3774cea20a365e1d959c8c865fdeacc82400da55e94ad800e75ba
   PATCHES
     disable-warning-nonnull.patch
     only-libsystemd.patch

--- a/ports/libsystemd/vcpkg.json
+++ b/ports/libsystemd/vcpkg.json
@@ -1,7 +1,6 @@
 {
   "name": "libsystemd",
-  "version": "255",
-  "port-version": 2,
+  "version": "256.2",
   "description": "Libsystemd",
   "homepage": "https://github.com/systemd/systemd",
   "license": null,

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5065,8 +5065,8 @@
       "port-version": 0
     },
     "libsystemd": {
-      "baseline": "255",
-      "port-version": 2
+      "baseline": "256.2",
+      "port-version": 0
     },
     "libtar": {
       "baseline": "1.2.20",

--- a/versions/l-/libsystemd.json
+++ b/versions/l-/libsystemd.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "9b6ef432aa434e0b1f188e23e6d1309a9050ab0b",
+      "version": "256.2",
+      "port-version": 0
+    },
+    {
       "git-tree": "4c173e2a6eaa6e20dae0122c0aee9ab9cfab82df",
       "version": "255",
       "port-version": 2

